R-package/R/lgb.Booster.R

```diff
@@ -207,6 +208,12 @@ Booster <- R6::R6Class(
     # Perform boosting update iteration
     update = function(train_set = NULL, fobj = NULL) {
 
+      if (is.null(train_set)) {
+        if (private$train_set$.__enclos_env__$private$version != private$train_set_version) {
```

Comment:
Instead of `x != y`, could you use `!identical(x, y)` here? The issue is that `having a `NULLL` on either side of a logical like this in R will cause an error.

![image](https://user-images.githubusercontent.com/7608904/71797150-77762f00-3012-11ea-9dc0-a48747c09611.png)

Reply:
given there are two integers, do we still need to use `identical`?

Reply:
I wasn't sure if they were guaranteed to always be non-null. You can use `!=`  if they will always be integers.

Reply:
yeah, if there is a NULL, it means there are bugs in the code.